compiler/src/test/java/com/squareup/anvil/compiler/dagger/AssistedFactoryGeneratorTest.kt

```diff
@@ -482,6 +482,51 @@ public final class AssistedServiceFactory_Impl implements AssistedServiceFactory
     }
   }
 
+  @Test fun `the factory function may be provided by a generic super type with generic parameter`() {
+
+    // This is broken for Dagger, but working with Anvil factory generation
+    if (useDagger) return
```

Comment:
I'm unsure about this. In the past I didn't support use cases that Dagger didn't support. I found problems with deeply nested `Provider<Provider<Provider<..>>>`. At least we should file a Dagger issue and link it here.

Reply:
My thinking is that if it works, then someone might be using it, so it's best to catch any regressions.  But I'm happy to remove the test if you'd prefer.

I'd intended to log an issue, but hadn't had a chance yet.  Here we are: https://github.com/google/dagger/issues/2984

I'll add a link to the comment.

Reply:
Keep the test. The link is good enough for me.